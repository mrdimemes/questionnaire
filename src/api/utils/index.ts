export { default as getFetchError } from "./getFetchError";