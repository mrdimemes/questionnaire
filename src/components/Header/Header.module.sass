@use "src/sass/foundation/colors"
@use "src/sass/foundation/breakpoints"
@use "src/sass/foundation/mixins"

.body
  background-color: colors.$main-light
  font-size: 2em
  position: fixed
  top: 0
  left: 0
  right: 0
  height: 60px

.wrapper
  height: 100%
  display: flex
  justify-content: space-between
  align-items: center

.logo
  font-weight: 900
  text-transform: uppercase
  @include mixins.unselectable

.interactive
  display: flex
  align-items: center
  column-gap: 45px

.navigation
  display: flex
  align-items: center
  column-gap: 30px

.link
  display: block

.activeLink
  font-weight: 800

@media (max-width: breakpoints.$small-screen)
  .body
    font-size: 1.8em
    height: 50px
