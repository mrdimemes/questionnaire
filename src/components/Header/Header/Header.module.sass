@use "src/sass/foundation/colors"
@use "src/sass/foundation/breakpoints"
@use "src/sass/foundation/mixins"

.body
  background-color: colors.$main-light
  font-size: 2em
  position: fixed
  top: 0
  left: 0
  right: 0
  height: 60px
  @media (max-width: breakpoints.$small-screen)
    font-size: 1.8em
    height: 50px
  @media (max-width: breakpoints.$tablet)
    font-size: 1.6em
    height: 40px

.wrapper
  display: flex
  justify-content: space-between
  align-items: center

.logo
  font-weight: 900
  text-transform: uppercase
  @include mixins.unselectable
