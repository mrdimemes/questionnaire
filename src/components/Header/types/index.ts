export type { HeaderProps } from "./HeaderProps";