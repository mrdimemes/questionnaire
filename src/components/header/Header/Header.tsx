import classNames from "classnames";
import { useThemeSelector } from "src/redux/hooks";
import { getThemeStyle } from "src/redux/slices/themeSlice";

import { Menu } from "../Menu";

import styles from "./Header.module.sass";

const Header = () => {
  const currentTheme = useThemeSelector();

  return (
    <header className={classNames(
      styles.body,
      getThemeStyle(styles, currentTheme),
    )}>
      <div className={`wrapper ${styles.wrapper}`}>
        <div className={styles.logo}>Questions!</div>
        <Menu />
      </div>
    </header>
  );
};

export default Header;