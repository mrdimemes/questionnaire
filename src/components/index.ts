export { App } from "./App";

export { Header } from "./header";

export { PaginationBar } from "./pagination";

export { TagComponent } from "./tags";
export { TagGallery } from "./tags";
export { TagBar } from "./tags";
export { AddTagWidget } from "./tags";

export { QuestionnaireGallery } from "./questionnaires/gallery";
export { QuestionnaireComponent } from "./questionnaires/questionnaire";
export { QuestionnaireEditor } from "./questionnaires/edit";

export { Button } from "./UI/Button";
export { Input } from "./UI/inputs";
export { CheckboxInput } from "./UI/inputs";
export { RadioInput } from "./UI/inputs";
export { TextInput } from "./UI/inputs";
export { EmailInput } from "./UI/inputs";
export { PasswordInput } from "./UI/inputs";
export { NameInput } from "./UI/inputs";
export { TextArea } from "./UI/inputs";
export { Select } from "./UI/inputs";

export { AuthForm } from "./AuthForm";

export { LoadingSpinner } from "./LoadingSpinner";