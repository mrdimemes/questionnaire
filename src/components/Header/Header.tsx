import classNames from "classnames";

import { withTheme } from "src/HOCs";

import { Menu } from "./Menu";
import styles from "./Header.module.sass";

import type { HeaderProps } from "./types";


const Header = ({ className }: HeaderProps) => {
  return (
    <header className={classNames(styles.Header, className)}>
      <div className={classNames(styles.wrapper, "wrapper")}>
        <div className={styles.logo}>Questions!</div>
        <Menu />
      </div>
    </header>
  );
};

export default withTheme(Header, styles);