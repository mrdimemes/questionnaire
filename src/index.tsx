import React from "react";
import ReactDOM from "react-dom/client";
import { BrowserRouter } from "react-router-dom";
import store from "./redux";
import { Provider } from "react-redux";
import { App } from "./components";
import "./sass/index.sass";

const root = ReactDOM.createRoot(
  document.getElementById("root") as HTMLElement
);
root.render(
  <BrowserRouter>
    <Provider store={store}>
      <App />
    </Provider>
  </BrowserRouter>
);
