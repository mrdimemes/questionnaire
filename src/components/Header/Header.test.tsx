import { screen } from "@testing-library/react";
import { renderWithProvidersAndRouters } from "src/utils/test-utils";
import { setupStore } from "src/redux";

import Header from "./Header";

describe("Header component", () => {
  test("Menu renders without user in the store", () => {
    const store = setupStore({ auth: { user: null } });
    renderWithProvidersAndRouters(<Header />, { store });
    expect(screen.getByText(/questions!/i)).toBeInTheDocument();
    expect(screen.getByRole("navigation")).toBeInTheDocument();
    expect(screen.getByText(/главная/i)).toBeInTheDocument();
    expect(screen.getByText(/тэги/i)).toBeInTheDocument();
    expect(screen.getByText(/опросы/i)).toBeInTheDocument();
    expect(screen.getAllByRole("button")).toHaveLength(3);
  });

  test("Menu renders with user in the store", () => {
    const store = setupStore({
      auth: {
        user: { name: "TestName", id: 0, isAdmin: false },
      },
    });
    renderWithProvidersAndRouters(<Header />, { store });
    expect(screen.getByText(/questions!/i)).toBeInTheDocument();
    expect(screen.getByText("TestName")).toBeInTheDocument();
    expect(screen.getByRole("navigation")).toBeInTheDocument();
    expect(screen.getByText(/главная/i)).toBeInTheDocument();
    expect(screen.getByText(/тэги/i)).toBeInTheDocument();
    expect(screen.getByText(/опросы/i)).toBeInTheDocument();
    expect(screen.getAllByRole("button")).toHaveLength(4);
  });

  test("Header snapshot", () => {
    const { container } = renderWithProvidersAndRouters(<Header />);
    expect(container).toMatchSnapshot();
  });
});