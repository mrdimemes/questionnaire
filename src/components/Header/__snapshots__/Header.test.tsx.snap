// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Header snapshot 1`] = `
<div>
  <header
    class="Header theme_light"
  >
    <div
      class="wrapper wrapper"
    >
      <div
        class="logo"
      >
        Questions!
      </div>
      <div
        class="Menu theme_light"
      >
        <button
          class="Button BurgerMenuButton theme_light theme_light"
        >
          <div
            class="icon"
          />
        </button>
        <div
          class="container"
        >
          <nav
            class="Navigation"
          >
            <a
              class="link"
              href="/"
            >
              Главная
            </a>
            <a
              class="link"
              href="/questionnaires"
            >
              Опросы
            </a>
            <a
              class="link"
              href="/tags"
            >
              Тэги
            </a>
          </nav>
          <div
            class="bar"
          >
            <button
              class="Button body theme_light theme_light"
            />
            <div
              class="UserPanel"
            >
              <button
                class="Button theme_light"
              >
                Войти
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </header>
</div>
`;
